assets/js/app.js

```diff
@@ -0,0 +1,443 @@
+    const initializeCheckbox = () => {
+        // Create all checkbox switches for all the coins
+        document.querySelectorAll("#coins-container .form-check-input").forEach(button => {
+            button.addEventListener("change", function () {
+                const selectedCoinsArray = getFromLocalStorage('selectedCoins') || [];
+                const coinId = this.id.replace('Switch', ""); // The id of the coin
+
+                if (this.checked === true) {
+                    // If checkbox is checked, handle coin selection
+                    if (selectedCoinsArray.length < 5) {
+                        // If less than 5 coins, add normally
+                        const allCoins = getFromLocalStorage('allCoins');
+                        const singleCoinData = allCoins.find(coin => coinId === coin.id);
+                        selectedCoinsArray.push(singleCoinData);
+                        saveToLocalStorage('selectedCoins', selectedCoinsArray);
+                    } else {
+                        // If trying to add a 6th coin, show modal
+                        moreThanFiveCoins(coinId, this);
+                    }
+                } else {
+                    // If unchecking, remove coin from selection
+                    const newCoinArray = selectedCoinsArray.filter(coin => coin.id !== coinId);
+                    saveToLocalStorage('selectedCoins', newCoinArray);
+                }
+            });
+        });
+    };
+
+    const loadSelectedButtons = () => {
+        const selectedCoinsArray = getFromLocalStorage('selectedCoins') || [];
+        selectedCoinsArray.forEach(coin => {
+            const checkbox = document.getElementById(`${coin.id + 'Switch'}`); // Get the checkbox element
+            if (checkbox) {
+                checkbox.checked = true; // Set the checkbox to checked
+            }
+        });
+    }
+
+    const handleCoinReplacement = (coinToReplaceId) => {
+        // Get current selected coins
+        const selectedCoins = getFromLocalStorage('selectedCoins') || [];
+        // Remove the coin to be replaced
+        const updatedSelectedCoins = selectedCoins.filter(coin => coin.id !== coinToReplaceId);
+
+        // Get the new coin's data and add it
+        const allCoins = getFromLocalStorage('allCoins');
+        const newCoinData = allCoins.find(coin => coin.id === pendingCoinId);
+        updatedSelectedCoins.push(newCoinData);
+
+        // Update localStorage
+        saveToLocalStorage('selectedCoins', updatedSelectedCoins);
+
+        // Update UI switches
+        const replacedCoinSwitch = document.getElementById(`${coinToReplaceId}Switch`);
+        if (replacedCoinSwitch) {
+            replacedCoinSwitch.checked = false;
+        }
+
+        // Reset pending coin data
+        pendingCoinId = null;
+        pendingCoinSwitch = null;
+    };
+
+    // this function is async because of the bootstrap modal
+    const moreThanFiveCoins = async (coinId, switchElement) => {
```

Comment:
do you know the shesh besh rule called `no more than five`? also referred to as the `egyptian rule`
doesn't have anything to do with the code but i had an association...